pilot/adapter/serviceregistry/aggregate/mesh_resource_set.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggregate
+
+import (
+	"math"
+)
+
+// Key used to track resources like service or service instance objects maintained by a registry
```

Comment:
For the record, I really don't see the rationale behind using terms like "resource", sets, views in an infrastructure where there are running services. Resources in mind evokes imagery related to storage/network/compute resource, or some static files, etc.